featuretools/feature_base/feature_base.py

```diff
@@ -275,7 +269,8 @@ def __rdiv__(self, other):
     def __mul__(self, other):
         """Multiply by other"""
         if isinstance(other, FeatureBase):
-            if self.variable_type == Boolean and other.variable_type == Boolean:
+            if (self.column_schema.logical_type == ltypes.Boolean and
+                    other.column_schema.logical_type == ltypes.Boolean):
```

Comment:
Not sure if it matters, but you could use `isinstance` here instead of checking equality:
`isinstance(self.column_schema.logical_type, ltypes.Boolean)`.

Adding the `is_boolean` property to WW would also help here...